### PR TITLE
Convert excel tests to support net6

### PR DESCRIPTION
**Pull request by marbelarrietaGlobant**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Convert DynamoMSOfficeTests project to support net6 conditionally in meanwhile WPF projects are migrated.  

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [x] This PR contains no files larger than 50 MB 

### Release Notes
DynamoMSOfficeTests project added to the Dynamo.All.Net6.sln

### CODE REVIEW

File: test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj [hunk 1]
```diff
@@ -11,16 +11,24 @@
...
     <AssemblyName>DynamoMSOfficeTests</AssemblyName>
   </PropertyGroup>
-  <ItemGroup>
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net48' ">
 	  <PackageReference Include="NUnit" Version="2.6.3" />
     <PackageReference Include="NUnitTestAdapter" Version="2.3.0" ExcludeAssets="all" />
       <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" />
     <Reference Include="PresentationCore" />
     <Reference Include="System" />
     <Reference Include="System.Core" />
```
Comment (aparajit-pratap) [inline test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj:20]: These are common to both net48 and net60 so can be listed outside of the two item groups.

File: test/Libraries/DynamoMSOfficeTests/ExcelTests.cs [hunk 2]
```diff
@@ -1362,10 +1364,12 @@ public void TestFormula()
         }     
         #endregion
     }
-
+#endif
     [TestFixture]
     public class CSVTests : UnitTestBase
     {
+#if NET6_0_OR_GREATER 
+#endif
```
Comment (aparajit-pratap) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs:1372]: ?

File: test/Libraries/DynamoMSOfficeTests/ExcelTests.cs [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
 namespace Dynamo.Tests
 {
-    [TestFixture, RequiresSTA]
+
+#if NET6_0_OR_GREATER
+    [TestFixture, Apartment(ApartmentState.MTA)]
+    public class ExcelTests
+    {
+    }
+#elif NETFRAMEWORK
```
Comment (aparajit-pratap) [inline test/Libraries/DynamoMSOfficeTests/ExcelTests.cs:17]: So effectively, you're excluding all tests from running in .net6?

File: test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj (hunk 1 above)
Comment (marbelarrietaGlobant) [reply test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj]: Solved

File: test/Libraries/DynamoMSOfficeTests/ExcelTests.cs (hunk 3 above)
Comment (marbelarrietaGlobant) [reply test/Libraries/DynamoMSOfficeTests/ExcelTests.cs:17]: Yes, I'm excluding all test because ExcelTests is inheriting from DynamoViewModelUnitTest class which is not yet present in Dynamo.All.Net6.sln

Comment (aparajit-pratap) [reply test/Libraries/DynamoMSOfficeTests/ExcelTests.cs:17]: Yes, I understand that. I was only trying to understand, then why start this PR in the first place until we first migrate the WPF projects or figure out a way to first decouple Excel tests from anything that had to do with WPF like inheriting from a model test class instead of `DynamoViewModelUnitTest`.

File: test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
 using System.Runtime.InteropServices;
+using System.Threading;
 
 // General Information about an assembly is controlled through the following 
 // set of attributes. Change these attribute values to modify the information
 // associated with an assembly.
 [assembly: AssemblyTitle("DynamoMSOfficeTests")]
 [assembly: AssemblyCulture("")]
+
+#if NET6_0_OR_GREATER
```
Comment (pinzart90) [inline test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs:12]: what is this about ?

**Comment by mjkkirschner** [comment]
looks like real github actions failures now, I tried to rerun them, and they seem to fail with real build errors now, not related to the NPM stuff from yesterday, please take a look @marbelarrietaGlobant.

File: test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (marbelarrietaGlobant) [reply test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs:12]: It is a condition needed for build test in net60 and net48.  Some NUnit tests need to run in STA threads.

**Comment by marbelarrietaGlobant** [comment]
> looks like real github actions failures now, I tried to rerun them, and they seem to fail with real build errors now, not related to the NPM stuff from yesterday, please take a look @marbelarrietaGlobant.

I added some missing usings @mjkkirschner

**Comment by mjkkirschner** [comment]
@marbelarrietaGlobant this branch has a conflict please fix it so the builds/tests can run.

**Comment by mjkkirschner** [comment]
nice @marbelarrietaGlobant - the builds seem to all pass - lets wait for the parallel tests to merge.

File: test/DynamoCoreTests/PackageDependencyTests.cs [hunk 5]
```diff
@@ -94,7 +94,7 @@ public void ZeroTouchPackageDependencyIsCollectedAndSerialized()
             Assert.AreEqual(package.Nodes.Select(n => n.ToString("N")), nodes);
         }
 
-        [Test,Category("FailureNET6")]
+        [Test,Category("WindowsOnly")]
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/PackageDependencyTests.cs:97]: @marbelarrietaGlobant - why have you changed these categories?

File: test/DynamoCoreTests/Nodes/DictionaryTests.cs [hunk 6]
```diff
@@ -79,7 +79,7 @@ public void EmptyDictionaryToString()
             Assert.AreEqual(dict.ToString(), "{}");
         }
 
-        [Test,Category("FailureNET6")]
+        [Test]
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/Nodes/DictionaryTests.cs:82]: @aparajit-pratap was this the same test that you fixed?

File: test/DynamoCoreTests/PackageDependencyTests.cs (hunk 5 above)
Comment (marbelarrietaGlobant) [reply test/DynamoCoreTests/PackageDependencyTests.cs:97]: According to the acceptance criteria of the US, the tests that failed in net60 were reviewed and when they were run it was found that they are not failing anymore in net60 for windows version. Linux continues failing.

File: test/DynamoCoreTests/Nodes/DictionaryTests.cs (hunk 6 above)
Comment (marbelarrietaGlobant) [reply test/DynamoCoreTests/Nodes/DictionaryTests.cs:82]: According to the acceptance criteria of the US, the tests that failed in net60 in this [PR](https://github.com/DynamoDS/Dynamo/pull/13766/files) were reviewed and when they were run it was found that they did not fail.

File: test/DynamoCoreTests/PackageDependencyTests.cs (hunk 5 above)
Comment (mjkkirschner) [reply test/DynamoCoreTests/PackageDependencyTests.cs:97]: @marbelarrietaGlobant I'd prefer if the changes in this PR are localized to just excel tests. 
Please leave these others as is - or add `WindowsOnly`, but keep the existing FailureNet6 in addition. Thanks.

File: test/DynamoCoreTests/Nodes/DictionaryTests.cs (hunk 6 above)
Comment (mjkkirschner) [reply test/DynamoCoreTests/Nodes/DictionaryTests.cs:82]: same request as above, the AC should have been more specific, it did not mean to refer to ALL tests in the repo.

Comment (aparajit-pratap) [reply test/DynamoCoreTests/Nodes/DictionaryTests.cs:82]: Yes, I think I had already removed the failure category from the Dictionary test.

File: test/DynamoCoreTests/PackageDependencyTests.cs (hunk 5 above)
Comment (marbelarrietaGlobant) [reply test/DynamoCoreTests/PackageDependencyTests.cs]: @mjkkirschner Ok I changed to FailureNET6 again to test in the future for Windows and Linux.

File: test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj [hunk 7]
```diff
@@ -11,16 +11,20 @@
...
+    <PackageReference Include="NUnitTestAdapter" Version="2.3.0" ExcludeAssets="all" /> 
   </ItemGroup>
+    <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' "> 
+        <PackageReference Include="NUnit" Version="3.13.3" ExcludeAssets="none" />
+        <PackageReference Include="NUnit3TestAdapter" Version="4.2.1" />   
+    </ItemGroup>
   <ItemGroup>
-    <ProjectReference Include="..\..\..\src\DynamoCoreWpf\DynamoCoreWpf.csproj">
+      <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.3.2" ExcludeAssets="none" />
+      <Reference Include="PresentationCore" />
```
Comment (pinzart90) [inline test/Libraries/DynamoMSOfficeTests/DynamoMSOfficeTests.csproj:24]: these references to PresentationCore, System and SystemCore should not be in dotnet6.

File: test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (pinzart90) [reply test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs:12]: But why is it in AssemblyInfo.cs ?
Is this a good place to put this code ?
Should we set the ApartmentState in the `OneTimeSetUp` code of the tests ?
Also how come `[assembly: RequiresSTA]` was not there before on the net48 target ? Why do we need it now. ?

Comment (marbelarrietaGlobant) [reply test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs]: You're right, I removed.

File: test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs [hunk 8]
```diff
@@ -1,10 +1,12 @@
-﻿using System.Reflection;
+using NUnit.Framework;
+using System.Reflection;
 using System.Runtime.InteropServices;
+using System.Threading;
```
Comment (pinzart90) [inline test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs:4]: I suggest you revert the changes to this file

Comment (pinzart90) [reply test/Libraries/DynamoMSOfficeTests/Properties/AssemblyInfo.cs]: @marbelarrietaGlobant please revert all changes to this file. 
I still see some whitespace changes.

**Review by pinzart90 — APPROVED** [review]
LGTM
Let's see what the PR checks say. Should be only 2 flaky UI tests
